solutions/kattis-catandmice.mdx

```diff
@@ -0,0 +1,105 @@
+---
+id: kattis-catandmice
+source: Kattis
+title: Cat and Mice
+author: Kevin Sheng
+---
+
+## Explanation
+
+Binary search on the minimum possible speed of Cartesian Cat.
+
+For each speed, let $\texttt{mintime}[S][e]$ be the minimum time to go from the start to end $e$, eating each mouse in the set $S$ along the way.
+If it is impossible to do so, the value will be defined as infinity.
+When processing each state, go through all the possible previous ending points and see if it's possible to reach the defined end given the previous ending time. Calculate the minimum time for the state over all valid previous endings.
+
+## Implementation
+
+<LanguageSection>
+<JavaSection>
+```java
+import java.io.BufferedReader;
```

Comment:
most java solutions just use java.util.* and java.io.* imports not extremely important but it would follow the usual format and look better imo